y/watermark.go

```diff
@@ -64,17 +64,13 @@ type WaterMark struct {
 	lastIndex uint64
 	Name      string
 	markCh    chan mark
-	elog      trace.EventLog
+	logger    badger.Logger
 }
 
 // Init initializes a WaterMark struct. MUST be called before using it.
-func (w *WaterMark) Init(closer *Closer, eventLogging bool) {
+func (w *WaterMark) Init(closer *Closer, opt badger.Options) {
```

Comment:
instead of passing a boolean value I am passing the Options object so that logger can be initialized for the Watermark struct as well.
@jarifibrahim, please see if this is the correct way to do it?

Reply:
this may cause the circular import error. I'm not clear about how would I do it otherwise. will creating the logger as a different module be a better approach?

Reply:
You can move the `logger` inside the `y` package. The `y` package contains a bunch of utility types.